Library/Homebrew/hardware.rb

```diff
@@ -214,6 +224,13 @@ def oldest_cpu(_version = nil)
       end
     end
     alias generic_oldest_cpu oldest_cpu
+
+    # Returns a _full_ rustflag to set target cpu, if necessary;
+    # Defaults to empty string
+    sig { returns(String) }
+    def rustflags_target_cpu
+      CPU.rust_optimisation_flags.fetch(oldest_cpu, "")
```

Comment:
```suggestion
    sig { returns(T.nilable(String)) }
    def rustflags_target_cpu
      CPU.rust_optimisation_flags.fetch(oldest_cpu, nil)
```

Also: we should try to support users doing something like
```sh
brew install --build-bottle --bottle-arch=ivybridge
```
which this doesn't currently do.